editor/editor_file_dialog.cpp

```diff
@@ -768,16 +833,33 @@ void EditorFileDialog::update_file_list() {
 		}
 
 		if (show_hidden_files || !dir_access->current_is_hidden()) {
-			if (!dir_access->current_is_dir()) {
-				files.push_back(item);
+			bool matches_search = false;
+			if (searched_string.length() > 0) {
+				matches_search = item.to_lower().find(searched_string) != -1;
 			} else {
-				dirs.push_back(item);
+				matches_search = true;
+			}
+			if (matches_search) {
+				if (!dir_access->current_is_dir()) {
+					FileInfo fi;
+					fi.name = item;
+					fi.path = cdir.plus_file(fi.name);
+					fi.type = ResourceLoader::get_resource_type(fi.path);
```

Comment:
I think `item.get_extension()` would be enough, `get_resource_type()` seems to use extension only too.